application/models/FleetModel.php

```diff
@@ -19,24 +19,31 @@ public function all()
     {
         $planes = parent::all();
         $planeIds = array_column($planes, 'airplaneCode');
-        $wackyPlanes = json_decode(file_get_contents(WACKY_SERVER_URI_BASE . '/airplanes'));
+        $wackyPlanes = $this->wackyModel->getAirplanes();
+
         $filteredPlanes = array_filter($wackyPlanes, function($p) use ($planeIds) {
             return in_array($p->id, $planeIds);
         });
 
+        $fleet = array();
+
         foreach ($filteredPlanes as $wackyPlane)
         {
             foreach ($planes as $plane)
             {
                 if ($plane->airplaneCode == $wackyPlane->id)
                 {
-                    foreach (get_object_vars($wackyPlane) as $prop => $val)
-                        if (!property_exists($plane, $prop))
-                            $plane->$prop = $val;
+                    $tempPlane = array();
+                    $tempPlane['id']           = $plane->id;
+                    $tempPlane['key']          = $plane->airplaneCode;
+                    $tempPlane['model']        = $wackyPlane->model;
+                    $tempPlane['manufacturer'] = $wackyPlane->manufacturer;
+                    array_push($fleet, $tempPlane);
```

Comment:
Why was this needed? Didn't the code you removed do the same thing?

Reply:
i wasn't sure what the previous code was doing so i tweaked it while testing something and forgot to undo. undid change.